src/qibolab/platform.py

```diff
@@ -29,104 +32,74 @@ class PlatformSettings:
-        self.settings: PlatformSettings = PlatformSettings(**settings["settings"])
-
-        # create qubit objects
-        self.qubits: Dict[QubitId, Qubit] = {
-            q: Qubit(q, **char) for q, char in settings["characterization"]["single_qubit"].items()
-        }
-
-        # create ``QubitPair`` objects
-        self.pairs: Dict[QubitPairId, QubitPair] = {}
-        for pair in settings["topology"]:
-            pair = tuple(sorted(pair))
-            self.pairs[pair] = QubitPair(self.qubits[pair[0]], self.qubits[pair[1]])
-
-        # register single qubit native gates to ``Qubit`` objects
-        native_gates = settings["native_gates"]
-        for q, gates in native_gates["single_qubit"].items():
-            self.qubits[q].native_gates = SingleQubitNatives.from_dict(self.qubits[q], gates)
-
-        # register two qubit native gates to ``QubitPair`` objects
-        self.two_qubit_native_types = NativeType(0)
-        if "two_qubit" in native_gates:
-            for pair, gatedict in native_gates["two_qubit"].items():
-                pair = tuple(sorted(int(q) if q.isdigit() else q for q in pair.split("-")))
-                self.pairs[pair].native_gates = TwoQubitNatives.from_dict(self.qubits, gatedict)
-                self.two_qubit_native_types |= self.pairs[pair].native_gates.types
-        else:
+    is_connected: bool = False
+    """Flag for whether we are connected to the physical instruments."""
+    two_qubit_native_types: NativeType = field(default_factory=lambda: NativeType(0))
+    """Types of two qubit native gates. Used by the transpiler."""
+    topology: nx.Graph = field(default_factory=nx.Graph)
+    """Graph representing the qubit connectivity in the quantum chip."""
+
+    def __post_init__(self):
+        log.info("Loading platform %s", self.name)
+        if self.resonator_type is None:
+            self.resonator_type = "3D" if self.nqubits == 1 else "2D"
+
+        for pair in self.pairs.values():
+            self.two_qubit_native_types |= pair.native_gates.types
+        if self.two_qubit_native_types is NativeType(0):
             # dummy value to avoid transpiler failure for single qubit devices
             self.two_qubit_native_types = NativeType.CZ
 
-        self.topology: nx.Graph = nx.Graph()
         self.topology.add_nodes_from(self.qubits.keys())
         self.topology.add_edges_from([(pair.qubit1.name, pair.qubit2.name) for pair in self.pairs.values()])
 
-    def __repr__(self):
+    def __str__(self):
         return self.name
 
-    def _check_connected(self):
-        if not self.is_connected:  # pragma: no cover
-            raise_error(RuntimeError, "Cannot access instrument because it is not connected.")
+    @property
+    def nqubits(self) -> int:
+        """Total number of usable qubits in the QPU.."""
+        # TODO: Seperate couplers from qubits (PR #508)
+        return len([qubit for qubit in self.qubits if not (isinstance(qubit, str) and "c" in qubit)])
```

Comment:
We should add a flag to identify couplers, not relying just on their name.

Reply:
Indeed, we started implementing a more proper solution in #508 so I went with a quick temporary fix here. 

The problem is that I am not very familiar with couplers as I have never used them (only one of our current chips has them), so I am not sure what would be a good object structure. That's why I would postpone.